packages/mdc-notched-outline/mdc-notched-outline.scss

```diff
@@ -28,57 +28,66 @@
 
 // Notched Outline is intended for use by multiple components, but its styles should only be emitted once when bundled
 @include mdc-base-emit-once("mdc-notched-outline") {
-  .mdc-notched-outline,
-  .mdc-notched-outline__idle {
+  .mdc-notched-outline {
+    display: flex;
     position: absolute;
-    top: 0;
+    right: 0;
     left: 0;
     box-sizing: border-box;
     width: 100%;
     height: 100%;
-  }
-
-  .mdc-notched-outline {
-    transition: opacity $mdc-notched-outline-transition-duration $mdc-animation-standard-curve-timing-function;
-    opacity: 0;
     text-align: left;
-    overflow: hidden;
+    pointer-events: none;
 
     @include mdc-rtl {
       /* @noflip */
       text-align: right;
     }
 
-    // stylelint-disable selector-max-type
-    svg {
-      position: absolute;
-      width: 100%;
+    &__leading,
+    &__notch,
+    &__trailing {
+      box-sizing: border-box;
       height: 100%;
+      border-top: 1px solid;
+      border-bottom: 1px solid;
+      pointer-events: none;
     }
-    // stylelint-enable selector-max-type
-  }
 
-  .mdc-notched-outline__idle {
-    transition:
-      opacity $mdc-notched-outline-transition-duration $mdc-animation-standard-curve-timing-function,
-      border-color $mdc-notched-outline-transition-duration $mdc-animation-standard-curve-timing-function;
-    border: 1px solid;
-    opacity: 1;
-  }
+    &__leading {
+      @include mdc-rtl-reflexive-property(border, 1px solid, none);
+
+      width: $mdc-notched-outline-leading-width;
+    }
 
-  .mdc-notched-outline__path {
-    stroke-width: 1px;
-    transition:
-      stroke $mdc-notched-outline-transition-duration $mdc-animation-standard-curve-timing-function,
-      stroke-width $mdc-notched-outline-transition-duration $mdc-animation-standard-curve-timing-function;
-    fill: transparent;
+    &__trailing {
+      @include mdc-rtl-reflexive-property(border, none, 1px solid);
+
+      flex-grow: 1;
+    }
+
+    &__notch {
+      width: auto;
+    }
+
+    .mdc-floating-label {
+      display: inline-block;
+      position: relative;
+      top: 17px;
+      bottom: auto;
+    }
   }
 
   .mdc-notched-outline--notched {
-    opacity: 1;
+    .mdc-notched-outline__notch {
+      padding: 0 8px 0 0;
+      border-top: none;
+    }
   }
 
-  .mdc-notched-outline--notched ~ .mdc-notched-outline__idle {
-    opacity: 0;
+  .mdc-notched-outline--no-label {
```

Comment:
Is this something we plan to detect and apply in JS? Do we need to document it anyway for FOUC prevention?

Would it alternatively be feasible to apply margin to the label instead, or would that be problematic due to the scaling?

Reply:
I'll update the notched outline to add this class if there is no label. 

I tried to use both padding and margin on the label but with the scale applied it's difficult to line up the label with the input and helper text.